lib/evaluate/fold.cc

```diff
@@ -295,6 +295,72 @@ static std::optional<std::int64_t> GetInt64ArgOr(
   }
 }
 
+template<typename A, typename B>
```

Comment:
It would, but the function is also used to extract other types than int64, and I would rather have to avoid having to convert std::vector<int64> to other integer types afterwards.